ServiceLayer/Services.cs

```diff
@@ -0,0 +1,133 @@
+﻿using DAL.Controllers;
+using DAL.Model;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ServiceLayer
+{
+    public class Services
+    {
+        Response response;
+        public Services()
+        {
+            response = new Response();
+        }
+        public Response GetById(int id, BookRepository bookRepository)
+        {
+            if (id < 0)
+            {
+                response.StatusCode = 400;
+                response.ErrorMessage = "Invalid Id, Id should be a positive number.";
+                return response;
+            }
+            var books = bookRepository.GetAllBooks();
+            var book = books.Find(x => x.Id == id);
+            if (book == null)
+            {
+                response.StatusCode = 404;
+                response.ErrorMessage = "Book Not Found";
+                return response;
+            }
+            else
+            {
+                response.StatusCode = 200;
+                response.Result = book;
+            }
+            return response;
+        }
+
+        public Response GetAllBooks(BookRepository bookRepository)
+        {
+            var books = bookRepository.GetAllBooks();
+            if(books.Count <=0)
+            {
+                response.StatusCode = 400;
+                response.ErrorMessage = "Books Not Found";
+                return response;
+            }
+            else
+            {
+                response.StatusCode = 200;
+                response.Result = books;
+                return response;
+            }
+        }
+
+        public Response PostByBook(Book book, BookRepository bookRepository)
+        {
+            if (book.Id < 0)
+            {
+                response.StatusCode = 400;
+                response.ErrorMessage = "Invalid Id, Id should be a positive number.";
+                return response;
```

Comment:
You can append multiple error messages here. Instead of returning response from here, you can append multiple status codes and error messages to the current response object and then send that at the end of the function